dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguageHandlerInterface.java

```diff
@@ -327,8 +333,13 @@ default String getURLFromEntry(String dockerEntry, ToolDAO toolDAO) {
 
         String url;
 
-        // Remove tag if exists
-        Pattern p = Pattern.compile("([^:]+):?(\\S+)?");
+        // Remove tag or digest if exists
+        Pattern p;
```

Comment:
Nit: ideally these would be static, which may be tricky with an interface. They are relatively expensive to create.